language-extensions/dotnet-core-CSharp/src/managed/utils/Logging.cs

```diff
@@ -41,7 +54,111 @@ public static void Trace(string message)
         /// </param>
         public static void Error(string message)
         {
-            Console.Error.WriteLine(message);
+            try
+            {
+                Console.Error.WriteLine(message);
+            }
+            catch (Exception)
+            {
+                // Ignore exceptions during error logging
+            }
+        }
+
+        /// <summary>
+        /// Custom TextWriter that writes directly to the standard output/error streams
+        /// using native system calls. This is required because the standard Console
+        /// output redirection might not work as expected in the embedded environment
+        /// where the .NET runtime is hosted by SQL Server.
+        /// </summary>
+        private class InteropTextWriter : System.IO.TextWriter
+        {
+            private readonly int _fd;
+
+            /// <summary>
+            /// Initializes a new instance of the InteropTextWriter class.
+            /// </summary>
+            /// <param name="fd">The file descriptor (1 for stdout, 2 for stderr).</param>
+            public InteropTextWriter(int fd)
+            {
+                _fd = fd;
+            }
+
+            /// <summary>
+            /// Gets the character encoding in which the output is written.
+            /// </summary>
+            public override Encoding Encoding => Encoding.UTF8;
+
+            /// <summary>
+            /// Writes a character to the text stream.
+            /// </summary>
+            /// <param name="value">The character to write.</param>
+            public override void Write(char value)
+            {
+                Write(value.ToString());
+            }
```

Comment:
Just FYI: I don't think this is performant. We allocate a new string for every character.